### ISSUE TITLE
Error in recipe for banana cream pie

### ISSUE THREAD

**Issue by shaneknysh** [ExcessiveAmountsOfZombies/Croptopia#441]
### Make sure you are not opening a duplicate.

- [X] [I have searched the issue tracker](https://github.com/ExcessiveAmountsOfZombies/Croptopia/issues?q=) and did not find an issue describing my bug.

### Platform.

Forge

### Minecraft version.

1.19.2

### Croptopia version.

2.1.0

### What happened?

During startup the error 
```
[03Sep.2022 16:05:22.908] [main/ERROR] [net.minecraft.world.item.crafting.RecipeManager/]: Parsing error loading recipe croptopia:banana_cream_pie
com.google.gson.JsonSyntaxException: Missing ingredient, expected to find a JsonObject
```

is displayed and the recipe for banana cream pie is not available in game. There is a related issue marked closed in version 1.3.4 (#90)

### Relevant log.

https://gist.github.com/shaneknysh/31643f15b50a0eec24c61b271aaef493

**Comment by archonsd** [comment]
FWIW, I have the same error in latest croptopia with latest fabric:

```
[09:26:08] [Render thread/ERROR]: Parsing error loading recipe croptopia:banana_cream_pie

com.google.gson.JsonSyntaxException: Missing ingredient, expected to find a JsonObject
	at net.minecraft.class_3518.method_15296(class_3518.java:413) ~[client-intermediary.jar:?]
	at net.minecraft.class_3957.method_17736(class_3957.java:28) ~[client-intermediary.jar:?]
	at net.minecraft.class_3957.method_8121(class_3957.java:11) ~[client-intermediary.jar:?]
	at net.minecraft.class_1863.method_17720(class_1863.java:135) ~[client-intermediary.jar:?]
	at net.minecraft.class_1863.method_20705(class_1863.java:56) ~[client-intermediary.jar:?]
	at net.minecraft.class_1863.method_18788(class_1863.java:35) ~[client-intermediary.jar:?]
	at net.minecraft.class_4080.method_18790(class_4080.java:13) ~[client-intermediary.jar:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:718) ~[?:?]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482) ~[?:?]
	at net.minecraft.class_4014.method_18365(class_4014.java:69) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.execute(class_1255.java:103) ~[client-intermediary.jar:?]
	at net.minecraft.class_4014.method_18367(class_4014.java:68) ~[client-intermediary.jar:?]
	at java.util.concurrent.CompletableFuture$UniCompletion.claim(CompletableFuture.java:572) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:714) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510) ~[?:?]
	at java.util.concurrent.CompletableFuture.postFire(CompletableFuture.java:614) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:726) ~[?:?]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482) ~[?:?]
	at net.minecraft.class_4014.method_18365(class_4014.java:69) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.execute(class_1255.java:103) ~[client-intermediary.jar:?]
	at net.minecraft.class_4014.method_18367(class_4014.java:68) ~[client-intermediary.jar:?]
	at java.util.concurrent.CompletableFuture$UniCompletion.claim(CompletableFuture.java:572) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:714) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510) ~[?:?]
	at java.util.concurrent.CompletableFuture.postFire(CompletableFuture.java:614) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:726) ~[?:?]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482) ~[?:?]
	at net.minecraft.class_4014.method_18365(class_4014.java:69) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.execute(class_1255.java:103) ~[client-intermediary.jar:?]
	at net.minecraft.class_4014.method_18367(class_4014.java:68) ~[client-intermediary.jar:?]
	at java.util.concurrent.CompletableFuture$UniCompletion.claim(CompletableFuture.java:572) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:714) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510) ~[?:?]
	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147) ~[?:?]
	at net.minecraft.class_4014$1.method_18374(class_4014.java:51) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.redirect$znd000$redirectExecuteTask(class_1255.java:521) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_16075(class_1255.java:131) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_18857(class_1255.java:140) ~[client-intermediary.jar:?]
	at net.minecraft.class_525.method_31130(class_525.java:158) ~[client-intermediary.jar:?]
	at net.minecraft.class_528.method_43462(class_528.java:147) ~[client-intermediary.jar:?]
	at net.minecraft.class_528.method_44680(class_528.java:107) ~[client-intermediary.jar:?]
	at net.minecraft.class_528$class_4272.method_33685(class_528.java:465) ~[client-intermediary.jar:?]
	at net.minecraft.class_528$class_4272.method_20170(class_528.java:445) ~[client-intermediary.jar:?]
	at net.minecraft.class_410.method_37054(class_410.java:55) ~[client-intermediary.jar:?]
	at net.minecraft.class_4185.method_25306(class_4185.java:43) ~[client-intermediary.jar:?]
	at net.minecraft.class_4264.method_25348(class_4264.java:16) ~[client-intermediary.jar:?]
	at net.minecraft.class_339.method_25402(class_339.java:120) ~[client-intermediary.jar:?]
	at net.minecraft.class_4069.method_25402(class_4069.java:27) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_1611(class_312.java:94) ~[client-intermediary.jar:?]
	at net.minecraft.class_437.method_25412(class_437.java:489) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_1601(class_312.java:94) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_22686(class_312.java:165) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.execute(class_1255.java:103) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_22684(class_312.java:165) ~[client-intermediary.jar:?]
	at org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:43) [lwjgl-glfw-3.3.1.jar:?]
	at org.lwjgl.system.JNI.invokeV(Native Method) ~[lwjgl-3.3.1.jar:?]
	at org.lwjgl.glfw.GLFW.glfwWaitEventsTimeout(GLFW.java:3474) [lwjgl-glfw-3.3.1.jar:?]
	at com.mojang.blaze3d.systems.RenderSystem.limitDisplayFPS(RenderSystem.java:222) [client-intermediary.jar:?]
	at net.minecraft.class_310.method_1523(class_310.java:1208) [client-intermediary.jar:?]
	at net.minecraft.class_310.method_1514(class_310.java:768) [client-intermediary.jar:?]
	at net.minecraft.client.main.Main.method_44604(Main.java:244) [client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:51) [client-intermediary.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) [fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.14.9.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at org.polymc.impl.OneSixLauncher.invokeMain(OneSixLauncher.java:104) [NewLaunch.jar:?]
	at org.polymc.impl.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:175) [NewLaunch.jar:?]
	at org.polymc.impl.OneSixLauncher.launch(OneSixLauncher.java:185) [NewLaunch.jar:?]
	at org.polymc.EntryPoint.listen(EntryPoint.java:144) [NewLaunch.jar:?]
	at org.polymc.EntryPoint.main(EntryPoint.java:74) [NewLaunch.jar:?]
```

**Comment by kingstairs** [comment]
Note that this error seems to block mod usage with the Quilt loader (current release and current beta) - Fabric seems to allow loading to proceed despite the exception, but Quilt shuts down completely.

**Comment by smmmadden** [comment]
appears to be an issue with the banana_cream_pie.json file missing an ingredient to the recipe, not loader specific.
[03Dec2022 08:36:14.649] [main/ERROR] [net.minecraft.world.item.crafting.RecipeManager/]: Parsing error loading recipe croptopia:banana_cream_pie
com.google.gson.JsonSyntaxException: Missing ingredient, expected to find a JsonObject
	at net.minecraft.util.GsonHelper.m_13930_(GsonHelper.java:413) ~[server-1.19.2-20220805.130853-srg.jar%23199!/:?]
	at net.minecraft.world.item.crafting.SimpleCookingSerializer.m_6729_(SimpleCookingSerializer.java:22) ~[server-1.19.2-20220805.130853-srg.jar%23199!/:?]
	at net.minecraft.world.item.crafting.SimpleCookingSerializer.m_6729_(SimpleCookingSerializer.java:11) ~[server-1.19.2-20220805.130853-srg.jar%23199!/:?]
	at net.minecraftforge.common.extensions.IForgeRecipeSerializer.fromJson(IForgeRecipeSerializer.java:23) ~[forge-1.19.2-43.1.65-universal.jar%23204!/:?]
	at net.minecraft.world.item.crafting.RecipeManager.fromJson(RecipeManager.java:171) ~[server-1.19.2-20220805.130853-srg.jar%23199!/:?]
	at net.minecraft.world.item.crafting.RecipeManager.m_5787_(RecipeManager.java:67) ~[server-1.19.2-20220805.130853-srg.jar%23199!/:?]
	at net.minecraft.world.item.crafting.RecipeManager.m_5787_(RecipeManager.java:34) ~[server-1.19.2-20220805.130853-srg.jar%23199!/:?]
	at net.minecraft.server.packs.resources.SimplePreparableReloadListener.m_10789_(SimplePreparableReloadListener.java:13) ~[server-1.19.2-20220805.130853-srg.jar%23199!/:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:718) ~[?:?]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482) ~[?:?]
	at net.minecraft.server.packs.resources.SimpleReloadInstance.m_143940_(SimpleReloadInstance.java:69) ~[server-1.19.2-20220805.130853-srg.jar%23199!/:?]
	at net.minecraft.Util.m_214652_(Util.java:781) ~[server-1.19.2-20220805.130853-srg.jar%23199!/:?]
	at net.minecraft.Util.m_214679_(Util.java:770) ~[server-1.19.2-20220805.130853-srg.jar%23199!/:?]
	at net.minecraft.server.Main.main(Main.java:157) ~[server-1.19.2-20220805.130853-srg.jar%23199!/:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at net.minecraftforge.fml.loading.targets.CommonServerLaunchHandler.lambda$launchService$0(CommonServerLaunchHandler.java:29) ~[fmlloader-1.19.2-43.1.65.jar%2367!/:?]
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) [modlauncher-10.0.8.jar%2354!/:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) [modlauncher-10.0.8.jar%2354!/:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) [modlauncher-10.0.8.jar%2354!/:?]
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:106) [modlauncher-10.0.8.jar%2354!/:?]
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:77) [modlauncher-10.0.8.jar%2354!/:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) [modlauncher-10.0.8.jar%2354!/:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) [modlauncher-10.0.8.jar%2354!/:?]
	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141) [bootstraplauncher-1.1.2.jar:?]

**Comment by Ayutac** [comment]
can confirm

**Comment by DelanoPreston** [comment]
I am getting the same error on forge

**Comment by ExcessiveAmountsOfZombies** [comment]
This should be fixed in 2.2.0
